athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/metadata/glue/GlueFieldLexer.java

```diff
@@ -59,99 +66,85 @@ default Field getField(String name, String type)
+                // Current token is not a child if it doesn't have a ":" (FIELD_DIV).
+                // Which means we are done with the struct. Just break out and build.
+                break;
             }
+            // Recursive call to resolve child type
+            Field child = lexInternal(childNameToken.getValue(), parser, mapper);
+            fieldBuilder.addField(child);
         }
-        parser.next();
-
-        logger.debug("lexComplex: exit - {}", name);
+        // Expect that we're ending on the closing token
+        expectTokenMarkerIsFieldEnd(parser.currentToken());
         return fieldBuilder.build();
     }
 
-    private static Field lex(GlueTypeParser.Token startToken, GlueTypeParser parser, BaseTypeMapper mapper)
+    private static Field parseList(String name, GlueTypeParser.Token typeToken, GlueTypeParser parser, BaseTypeMapper mapper)
     {
-        GlueTypeParser.Token nameToken = startToken;
-        logger.debug("lex: enter - {}", nameToken.getValue());
-        if (!nameToken.getMarker().equals(GlueTypeParser.FIELD_DIV)) {
-            throw new RuntimeException("Expected Field DIV but found " + nameToken.getMarker() +
-                    " while processing " + nameToken.getValue());
-        }
-
-        String name = nameToken.getValue();
+        expectTokenMarkerIsFieldStart(typeToken);
+        // Recursive call to resolve child element type
+        Field child = lexInternal(name, parser, mapper);
+        // The next field must always be a closing token since we are building the field here
+        // So consume the closing token for this field
+        GlueTypeParser.Token closingToken = parser.next();
+        // Note that the closing token is , if the enclosing type is a struct
+        // and > if the enclosing type is a list
+        expectTokenMarkerIsFieldEnd(closingToken);
+        return FieldBuilder.newBuilder(name, Types.MinorType.LIST.getType()).addField(child).build();
+    }
 
-        GlueTypeParser.Token typeToken = parser.next();
-        if (typeToken.getMarker().equals(GlueTypeParser.FIELD_START)) {
-            logger.debug("lex: exit - {}", nameToken.getValue());
-            return lexComplex(name, typeToken, parser, mapper);
+    private static void expectTokenMarkerIsFieldStart(GlueTypeParser.Token token)
+    {
+        if (token.getMarker() != GlueTypeParser.FIELD_START) {
+            throw new RuntimeException("Expected field start: but found " + token.toString());
         }
-        else if (typeToken.getMarker().equals(GlueTypeParser.FIELD_SEP) ||
-                typeToken.getMarker().equals(GlueTypeParser.FIELD_END)
-        ) {
-            logger.debug("lex: exit - {}", nameToken.getValue());
-            return mapper.getField(name, typeToken.getValue());
+    }
+
+    private static void expectTokenMarkerIsFieldEnd(GlueTypeParser.Token token)
+    {
+        boolean isFieldEnd = (token.getMarker() == null ||
```

Comment:
why would null be field end? Isn't it going to come with GlueTypeParser.FIELD_END?

Reply:
FIeld end being null happens at the very end of the type string.